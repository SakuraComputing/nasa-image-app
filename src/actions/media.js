import { SET_MEDIA, GET_MEDIA } from '../actions/types';
import axios from 'axios';

export const setMedia = (media) => ({
    type: SET_MEDIA,
    media
});

export const getMedia = (searchString, media_type) => dispatch => {

    let searchUri;

    if(media_type) {
        searchUri = `https://images-api.nasa.gov/search?q=${searchString}&media_type=${media_type}`;
    } else {
        searchUri = `https://images-api.nasa.gov/search?q=${searchString}`;
    }
    axios.get(searchUri)
    .then(res => 
        dispatch({
            type: GET_MEDIA,
            payload: res.data
        })
    )
    .catch(err => console.log(err));
}
