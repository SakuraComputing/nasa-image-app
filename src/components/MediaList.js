import React from 'react'

const MediaList = ({ title, item }) => {
  return (
    <div className="media-box image-frame">
      <h3>{title}</h3>
      {item.links 
        ? <img className="image-item" src={item.links[0].href} alt="noimage"/>
        : <p>{item.data[0].description}</p>}
    </div>
  )
}
export default MediaList;


