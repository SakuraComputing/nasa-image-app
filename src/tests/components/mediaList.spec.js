import React from 'react';
import { shallow } from 'enzyme';
import MediaList from '../../components/MediaList';
import media from '../fixtures/media';

describe('MediaList component', () => {
    it('should render correctly', () => {
        const item = { 
            links: [ { href: jest.fn() } ],
            data: [ { nasa_id: jest.fn() }]
        };
        const mediaList = shallow(<MediaList item={item} {...media[0]}/>)
        expect(mediaList).toMatchSnapshot();
    });
});