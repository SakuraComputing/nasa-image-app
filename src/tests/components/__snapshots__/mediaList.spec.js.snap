// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaList component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaList
    description="Voyage to the moon"
    image="https://images.assets.nasa.gov/apollo11"
    item={
      Object {
        "links": Array [
          Object {
            "href": [MockFunction],
          },
        ],
      }
    }
    title="Apollo 11"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Apollo 11
        </h3>,
        <img
          alt="noimage"
          className="image-item"
          src={[MockFunction]}
        />,
      ],
      "className": "media-box image-frame",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Apollo 11",
        },
        "ref": null,
        "rendered": "Apollo 11",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "noimage",
          "className": "image-item",
          "src": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Apollo 11
          </h3>,
          <img
            alt="noimage"
            className="image-item"
            src={[MockFunction]}
          />,
        ],
        "className": "media-box image-frame",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Apollo 11",
          },
          "ref": null,
          "rendered": "Apollo 11",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "noimage",
            "className": "image-item",
            "src": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
